### DISCUSSION TITLE
동적 생성되는 캐릭터의 카메라를 가져오고 싶습니다.

### DISCUSSION THREAD

**Discussion by smartbh** [naverz/zepeto-studio-kor#71] [5. Q&A]
버튼을 사용하기 위해서 월드 스페이스의 이벤트 카메라가 필요합니다

동적 생성되는 플레이어의 카메라를 이용하기 위해서 코드를 짰는데

잘 작동이 되질 않아 문의드립니다.

![image](https://user-images.githubusercontent.com/43138813/145730137-d34fafb4-5d50-451f-a328-dc1243c58320.png)

적용을 시키려는 캔버스의 인스펙터 상황이고

![image](https://user-images.githubusercontent.com/43138813/145730178-68c2b07e-da4c-4247-9d32-fff9bbdfb660.png)

캔버스에 할당 된 스크립트의 상태입니다.

어떻게든 해보려고 여러 방법으로 짜봤는데 잘 되질 않아

흔적들이 좀 많이 남아있습니다.

뭔가 놓친부분이 있을까요?

**Comment by mhkim0720** [comment]
this.cam = GameObject.Find('CameraParent').GetComponentInChildren<카메라>();

**Reply by smartbh** [reply]
아 감사합니다. 이 방법도 안해봤던것은 아닙니다 this.PlayerCam = UnityEngine.GameObject.Find('CameraParent').GetComponentInChildren<UnityEngine.Camera>() 이렇게 해서 받았을때도

저번에 올렸던 Getcomponent나 GetcomponentInChildren 프로퍼티를 가져올수 없다는 Ts null 오류가 떠서요

**Reply by smartbh** [reply]
아 그리고 만약에 받았다고 쳐도 Canvas의 이벤트 카메라에 할당이 안되는것 같습니다. 현재 최종 수정한 코드가

private PlayerCam: UnityEngine.Camera; 
public canv : UnityEngine.Canvas; 라고 선언하고

player캠은 
this.PlayerCam = UnityEngine.GameObject.Find('CameraParent').GetComponentInChildren<UnityEngine.Camera>();로 잘받아왔는데

이걸 캔버스의 EventCamera에 배정해 주려고

this.PlayerCam = UnityEngine.GameObject.Find('CameraParent').GetComponentInChildren<UnityEngine.Camera>();
this.canv.worldCamera = this.PlayerCam;
console.log("Camera worked!!");

이렇게 다시 짜봤는데

유니티 에디터 상에서 런타임으로 실행시켜보면

콘솔로그 창에 Camera worked뜨면서 오류는 더이상 나지 않는데

여전히 이벤트 카메라는 None으로 배정이 되질않네요 흠..

혹시 canvas.worldcamera가 아니라 다른곳에 넣어줘야 하나요? 어제 하위 함수 하나하나 전부 확인해 보긴 했는데

유니티 도큐멘트에서도 저 worldcamera가 이벤트 카메라라고 돼있었거든요

**Reply by smartbh** [reply]
해결했습니다.. 진짜.. 말도 안되는 문제였습니다.. 어이가 없네요 말도 안되는 고민 계속 들어주셔서 정말 감사합니다..
